tests/api/api/test_runs.py

```diff
@@ -338,6 +340,71 @@ def test_list_runs_partition_by(db: Session, client: TestClient) -> None:
     assert response.status_code == HTTPStatus.UNPROCESSABLE_ENTITY.value
 
 
+def test_delete_runs_with_permissions(db: Session, client: TestClient):
+    mlrun.api.utils.auth.verifier.AuthVerifier().query_project_resource_permissions = (
+        unittest.mock.Mock()
+    )
+
+    # delete runs from specific project
+    project = "some-project"
+    _store_run(db, uid="some-uid", project=project)
+    runs = mlrun.api.crud.Runs().list_runs(db, project=project)
+    assert len(runs) == 1
+    response = client.delete(f"/api/runs?project={project}")
+    assert response.status_code == HTTPStatus.OK.value
+    runs = mlrun.api.crud.Runs().list_runs(db, project=project)
+    assert len(runs) == 0
+
+    # delete runs from all projects
+    second_project = "some-project2"
+    _store_run(db, uid=None, project=project)
+    _store_run(db, uid=None, project=second_project)
+    all_runs = mlrun.api.crud.Runs().list_runs(db, project="*")
+    assert len(all_runs) == 2
+    response = client.delete("/api/runs?project=*")
+    assert response.status_code == HTTPStatus.OK.value
+    runs = mlrun.api.crud.Runs().list_runs(db, project="*")
+    assert len(runs) == 0
+
+
+def test_delete_runs_without_permissions(db: Session, client: TestClient):
+    mlrun.api.utils.auth.verifier.AuthVerifier().query_project_resource_permissions = (
+        unittest.mock.Mock(side_effect=mlrun.errors.MLRunUnauthorizedError())
+    )
+
+    # try delete runs with no permission to project (project doesn't contain any runs)
+    project = "some-project"
+    runs = mlrun.api.crud.Runs().list_runs(db, project=project)
+    assert len(runs) == 0
+    response = client.delete(f"/api/runs?project={project}")
+    assert response.status_code == HTTPStatus.UNAUTHORIZED.value
+
+    # try delete runs with no permission to project (project contains runs)
+    _store_run(db, uid="some-uid", project=project)
+    runs = mlrun.api.crud.Runs().list_runs(db, project=project)
+    assert len(runs) == 1
+    response = client.delete(f"/api/runs?project={project}")
+    assert response.status_code == HTTPStatus.UNAUTHORIZED.value
+    runs = mlrun.api.crud.Runs().list_runs(db, project=project)
+    assert len(runs) == 1
+
+    # try delete runs from all projects with no permissions
+    response = client.delete("/api/runs?project=*")
+    assert response.status_code == HTTPStatus.UNAUTHORIZED.value
+    runs = mlrun.api.crud.Runs().list_runs(db, project=project)
+    assert len(runs) == 1
```

Comment:
I'm not 100% sure about this, tell me what you think: Instead of having a test for `with_perms` and one for `without_perms`, have 2 tests: `test_list_runs_permissions` and `test_delete_runs_permissions`, each of them will use `pytest.mark.parametrize` for stating wether you have permissions or not, and will know by that how to set the mock, check the status codes and the expected results.
I think this will make for a more robust coverage and will be easier to check other functionalities in the future separately from these tests.

Reply:
Tried that option it complicated the code, decided to leave it.